include/univalue.h

```diff
@@ -166,10 +169,24 @@ class UniValue {
     // value is of unexpected type
     const std::vector<std::string>& getKeys() const;
     const std::vector<UniValue>& getValues() const;
+    template <typename Int>
+    auto getInt() const
+    {
+        static_assert(std::is_integral<Int>::value);
+        if (typ != VNUM) {
+            throw std::runtime_error("JSON value is not an integer as expected");
+        }
+        Int result;
+        const auto [first_nonmatching, error_condition] = std::from_chars(val.data(), val.data() + val.size(), result);
+        if (first_nonmatching != val.data() + val.size() || error_condition != std::errc{}) {
+            throw std::runtime_error("JSON integer out of range");
+        }
+        return result;
+    }
     bool get_bool() const;
     const std::string& get_str() const;
-    int get_int() const;
-    int64_t get_int64() const;
+    auto get_int() const { return getInt<int>(); };
+    auto get_int64() const { return getInt<int64_t>(); };
```

Comment:
While here, I've occasionally wished there was a `get_uint64` helper, to avoid casting uint64 to int64 in the RPC code.

Reply:
I am adding getInt<uint64_t> in this very pull request

Reply:
Ah, this allows `getInt<uint64_t>()`; maybe describe the interface briefly in the pull description.

Reply:
Changed title